### PR TITLE
[FLINK-29774] Introduce options metadata table

### PR DESCRIPTION
**Pull request by JingsongLi**

### CODE REVIEW

File: docs/content/docs/development/query-table.md [hunk 1]
```diff
@@ -109,3 +109,19 @@ SELECT * FROM MyTable$snapshots;
 
 By querying one table's snapshots table, you can know the commit and expiration
 information about that table and time travel through the data.
+
+## Options Table
+
+You can query the option information of the table through Flink SQL. The options not
```
Comment (LadyForest) [inline docs/content/docs/development/query-table.md:115]: How about 
> You can query the table's option information which is specified from the DDL through Flink SQL. The options not shown will be the default value.

File: flink-table-store-core/src/main/java/org/apache/flink/table/store/table/metadata/OptionsTable.java [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+
+    @Override
+    public TableRead newRead() {
+        return new OptionsRead();
+    }
+
+    private class OptionsScan implements TableScan {
+
+        @Override
+        public TableScan withFilter(Predicate predicate) {
```
Comment (LadyForest) [inline flink-table-store-core/src/main/java/org/apache/flink/table/store/table/metadata/OptionsTable.java:92]: add a TODO?

File: flink-table-store-core/src/main/java/org/apache/flink/table/store/table/metadata/OptionsTable.java [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+        }
+    }
+
+    private static class OptionsRead implements TableRead {
+
+        private int[][] projection;
+
+        @Override
+        public TableRead withFilter(Predicate predicate) {
+            return this;
```
Comment (LadyForest) [inline flink-table-store-core/src/main/java/org/apache/flink/table/store/table/metadata/OptionsTable.java:145]: Add TODO?

File: flink-table-store-core/src/main/java/org/apache/flink/table/store/table/metadata/OptionsTable.java (hunk 2 above)
Comment (JingsongLi) [reply flink-table-store-core/src/main/java/org/apache/flink/table/store/table/metadata/OptionsTable.java:92]: No, I think we don't need to implement predicate pushdown for this.

File: flink-table-store-core/src/main/java/org/apache/flink/table/store/table/metadata/OptionsTable.java [hunk 4]
```diff
@@ -0,0 +1,182 @@
...
+import java.util.Collections;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.Objects;
+
+import static org.apache.flink.table.store.file.catalog.Catalog.METADATA_TABLE_SPLITTER;
+import static org.apache.flink.table.store.file.utils.SerializationUtils.newStringType;
+
+/** A {@link Table} for showing options of table. */
+public class OptionsTable implements Table {
```
Comment (LadyForest) [inline flink-table-store-core/src/main/java/org/apache/flink/table/store/table/metadata/OptionsTable.java:51]: I feel `OptionsTable` shares almost the same logic as `SnapshotsTable`. Do you consider adding an abstract base class now or in the future?

**Review by LadyForest — COMMENTED** [review]
Thanks for the contribution, and it looks good to me in general. I left some minor comments.

File: flink-table-store-core/src/main/java/org/apache/flink/table/store/table/metadata/OptionsTable.java (hunk 4 above)
Comment (JingsongLi) [reply flink-table-store-core/src/main/java/org/apache/flink/table/store/table/metadata/OptionsTable.java:49]: I have tried, but it is so simple, actually there are 4 classes, each class is very simple.
Add 4 abstract classes can not share much codes.

**Review by LadyForest — APPROVED** [review]
Thanks for the update. LGTM

**Pull request merged**